### PR TITLE
#916@patch: Handle comments with dash in text.

### PR DESCRIPTION
**Pull request by gtm-nayan**

### CODE REVIEW

File: packages/happy-dom/src/xml-parser/XMLParser.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ import * as Entities from 'entities';
  * Group 8: End of start tag (e.g. ">" in "<div>").
  */
 const MARKUP_REGEXP =
-	/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([^-]+)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm;
+	/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([^-]+(?:-[^-]+)*)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm;
```
Comment (capricorn86) [inline packages/happy-dom/src/xml-parser/XMLParser.ts:27]:
```suggestion
	/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([\s\S]+)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm;
```

Comment (capricorn86) [reply packages/happy-dom/src/xml-parser/XMLParser.ts:27]: My suggestion is that we just capture all characters until "-->" instead.

**Review by capricorn86 — CHANGES_REQUESTED** [review]

File: packages/happy-dom/src/xml-parser/XMLParser.ts (hunk 1 above)
Comment (capricorn86) [reply packages/happy-dom/src/xml-parser/XMLParser.ts]: I have commited my suggestion now as I would like to release a fix as soon as possible :sweat_smile:

**Comment by capricorn86** [comment]
Thank you for your contribution @gtm-nayan! :star2:

File: packages/happy-dom/src/xml-parser/XMLParser.ts (hunk 1 above)
Comment (capricorn86) [reply packages/happy-dom/src/xml-parser/XMLParser.ts]: That suggestion did not work. The problem is that the RegExp you suggested doesn't cover when there multiple dashes (e.g. "").

I'm looking into another solution.

Comment (gtm-nayan) [reply packages/happy-dom/src/xml-parser/XMLParser.ts]: I think making it non-greedy should work so 

`/<([a-zA-Z0-9-]+)|<\/([a-zA-Z0-9-]+)>|<!--([\s\S]+?)-->|<!--([^>]+)>|<!([^>]+)>|<\?([^>]+)>|(\/>)|(>)/gm`

**Comment by capricorn86** [comment]
@gtm-nayan I have changed the logic so that it will work with comments containing multiple dashes (e.g. `<!-- comment -a-b-c -->`).

**Review by capricorn86 — APPROVED** [review]

**Comment by gtm-nayan** [comment]
Thanks for the fix.

**Pull request merged**